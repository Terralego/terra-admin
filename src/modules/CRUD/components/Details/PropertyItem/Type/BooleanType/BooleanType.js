import React from 'react';
import PropTypes from 'prop-types';
import NoValue from '../../NoValue';

const BooleanType = ({
  display_value: displayValue,
  t,
}) => {
  if (displayValue === null) {
    return <NoValue />;
  }
  return (
    displayValue
      ? t('CRUD.details.true')
      : t('CRUD.details.false')
  );
};

BooleanType.propTypes = {
  display_value: PropTypes.bool,
  t: PropTypes.func,
};

BooleanType.defaultProps = {
  display_value: null,
  t: text => text,
};

export default BooleanType;
