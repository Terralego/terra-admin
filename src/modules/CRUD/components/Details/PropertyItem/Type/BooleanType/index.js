import { withTranslation } from 'react-i18next';
import compose from '../../../../../../../utils/compose';

import BooleanType from './BooleanType';

export default compose(
  withTranslation(),
)(BooleanType);
