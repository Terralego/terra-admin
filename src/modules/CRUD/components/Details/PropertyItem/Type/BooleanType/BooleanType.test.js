import React from 'react';
import renderer from 'react-test-renderer';
import BooleanType from './BooleanType';

jest.mock('../../NoValue', () => () => <div>No value</div>);

describe('should render correctly', () => {
  it('true', () => {
    const tree = renderer.create((
      <BooleanType display_value />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('false', () => {
    const tree = renderer.create((
      <BooleanType display_value={false} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('nothing', () => {
    const tree = renderer.create((
      <BooleanType display_value={null} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
