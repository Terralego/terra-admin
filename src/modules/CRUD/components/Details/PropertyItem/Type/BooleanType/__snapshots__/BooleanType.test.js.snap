// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly false 1`] = `"CRUD.details.false"`;

exports[`should render correctly nothing 1`] = `
<div>
  No value
</div>
`;

exports[`should render correctly true 1`] = `"CRUD.details.true"`;
